.github/workflows/master.yml

```diff
@@ -379,35 +379,22 @@ jobs:
           # files created during tests on Windows as it keeps the DB connection open until
           # it's explicitly disposed.
           MLFLOW_SQLALCHEMYSTORE_POOLCLASS: "NullPool"
-        run: |
           # Set Hadoop environment variables required for testing Spark integrations on Windows
-          export HADOOP_HOME=/tmp/winutils/hadoop-3.2.2
+          HADOOP_HOME: /tmp/winutils/hadoop-3.2.2
+        run: |
           export PATH=$PATH:$HADOOP_HOME/bin
-
-          # Define function to run pytest with common arguments
-          run_pytest() {
-            uv run --no-sync pytest "$@" \
-              --ignore-flavors \
-              --ignore=tests/projects \
-              --ignore=tests/examples \
-              --ignore=tests/evaluate \
-              --ignore=tests/optuna \
-              --ignore=tests/pyspark/optuna \
-              --ignore=tests/genai \
-              --ignore=tests/sagemaker \
-              --ignore=tests/gateway \
-              --ignore=tests/server/auth \
-              --ignore=tests/data/test_spark_dataset.py \
-              tests
-          }
-
-          # Run Windows tests
-          # Retry failed tests once to handle flaky tests on Windows
-          set +e
-          run_pytest --splits=${{ matrix.splits }} --group=${{ matrix.group }}
-          status=$?
-          set -e
-          if [ $status -ne 0 ]; then
-            echo "Retrying failed tests..."
-            run_pytest --last-failed
-          fi
```

Comment:
This approach produces incorrect test results. https://github.com/mlflow/mlflow/actions/runs/20339404970/job/58434336408 is an example